ReactCommon/yoga/yoga.podspec

```diff
@@ -47,7 +47,7 @@ Pod::Spec.new do |spec|
   source_files = File.join('ReactCommon/yoga', source_files) if ENV['INSTALL_YOGA_WITHOUT_PATH_OPTION']
   spec.source_files = source_files
 
-  header_files = 'yoga/{Yoga,YGEnums,YGMacros,YGValue,YGStyle,CompactValue,YGFloatOptional,Yoga-internal,YGNode,YGConfig,YGLayout,YGMarker}.h'
+  header_files = 'yoga/{Yoga,YGEnums,YGMacros,YGValue}.h'
```

Comment:
What's going on with this change?

Reply:
Ah, found the discussion at the top

Reply:
This reverts a change that was made in https://github.com/facebook/react-native/pull/23803 which admittedly I don't have full context on.

The reason it is needed is that with all of the headers public, the module fails to build when imported into `RCTImageSource.m` with the error below because the headers include C++ code.

```
▸ Compiling RCTImageSource.m

❌  /Users/james/src/tmp/AwesomeProject/node_modules/react-native/ReactCommon/yoga/yoga/CompactValue.h:11:10: 'cmath' file not found

#include <cmath>
           ^~~~~~~~~~~~~~~~~~~~~~~~~



❌  /Users/james/src/tmp/AwesomeProject/node_modules/react-native/React/Base/RCTConvert.h:17:9: could not build module 'yoga'

#import <yoga/Yoga.h>
         ^~~~~~~
```